app/Services/UserService.php

```diff
@@ -122,4 +126,39 @@ public function loginWithGoogle($data)
         return ['user' => $user, 'token' => $token];
     }
 
+    public function requestPasswordReset($email)
+    {
+        $user = User::where('email', $email)->first();
+        if (!$user) {
+            return response()->json(['error' => 'User not found'], 404);
+        }
+
```

Comment:
first you should make a query if that user already has an existing token and if the token has expired make the expirie time 5 minutes